src/tile.cpp

```diff
@@ -310,16 +307,16 @@ Item* Tile::getItemByTopOrder(int32_t topOrder)
 		          end = ItemVector::const_reverse_iterator(items->getBeginTopItem());
 		     it != end; ++it) {
 			if (Item::items[(*it)->getID()].alwaysOnTopOrder == topOrder) {
-				return (*it);
+				return *it;
 			}
 		}
 	}
 	return nullptr;
 }
 
-Thing* Tile::getTopVisibleThing(const Creature* creature)
+std::shared_ptr<Thing> Tile::getTopVisibleThing(std::shared_ptr<const Creature> creature)
 {
-	Thing* thing = getTopVisibleCreature(creature);
+	auto thing = getTopVisibleCreature(creature);
```

Comment:
Could this be `const auto`
```suggestion
	const auto thing = getTopVisibleCreature(creature);
```